models/classes/runner/QtiRunnerService.php

```diff
@@ -1317,7 +1325,7 @@ public function switchClientStoreId(RunnerServiceContext $context, $receivedStor
             $session = $context->getTestSession();
             $sessionId = $session->getSessionId();
 
-            $stateService = new ExtendedStateService();
+            $stateService = $this->getServiceLocator()->get(ExtendedStateService::SERVICE_ID);
```

Comment:
can you use getServiceManager() instead of getServiceLocator() to keep consistency